File: Data Structures & Algorithms/missing-number/submission-3.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        n = n*(n+1)/2;
        for(int x:nums){
            n-=x;
        }
        return n;
        int xr=n;
        for(int i=0;i<n;i++){
            xr = xr^i;
            xr = xr^(nums[i]);
        }
        return xr;
        return n;
    }
};

File: Data Structures & Algorithms/sort-colors/submission-3.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        mid:int = 0
        start:int = 0
        end:int = len(nums)-1

        while mid<=end:
            if nums[mid]==0:
                nums[mid],nums[start]=nums[start],nums[mid]
                start+=1
            elif nums[mid]==2:
                nums[mid],nums[end]=nums[end],nums[mid]
                end-=1
                mid-=1
            mid+=1
            
        

        